src/keyDerivation.c

```diff
@@ -0,0 +1,75 @@
+#include <os_io_seproxyhal.h>
+#include <stdint.h>
+
+#include "assert.h"
+#include "errors.h"
+#include "keyDerivation.h"
+
+#define VALIDATE_PARAM(cond) if (!(cond)) THROW(ERR_INVALID_REQUEST_PARAMETERS)
+
+void validatePath(uint32_t* path, uint32_t pathLength)
+{
+	uint32_t bip44 = BIP_44 | HARDENED_BIP32;
+	uint32_t adaCoinType = ADA_COIN_TYPE | HARDENED_BIP32;
+
+	VALIDATE_PARAM(pathLength >= 3 && pathLength <= 10);
+
+	VALIDATE_PARAM(path[0] ==  bip44);
+	VALIDATE_PARAM(path[1] ==  adaCoinType);
+	VALIDATE_PARAM(path[2] >= HARDENED_BIP32);
+}
+
+void derivePrivateKey(
+        uint32_t* bip32Path, uint32_t pathLength,
+        uint8_t* chainCode,
+        cx_ecfp_256_extended_private_key_t* privateKey
+)
+{
+	validatePath(bip32Path, pathLength);
+
+	uint8_t privateKeyRaw[64];
+
+	STATIC_ASSERT(CX_APILEVEL >= 5, unsupported_api_level);
+	os_memset(chainCode, 0, 32);
+
+	os_perso_derive_node_bip32(
```

Comment:
Hmm, this doesn't take 64 anywhere? How do we make sure we are getting correct lengths? ;-)

Reply:
hmm, if we really want to be paranoid, we would add sentinel and check value :-) I am not sure if that would be worth the trouble though ...